JavaWithMaven / POM Project

vieiro opened the pull request:
hotfix for #3809 
- Creates a maven template with a `<packaing>pom</packaging>` pom.xml

neilcsmith-net commented:
Glad to see that one fixed!  It might be good to get this fix in NB14 given it looks like we're having another RC?  What do you think?  Could you rebase on delivery?  That way can merge into either as need be.

mbien commented on java/maven/src/org/netbeans/modules/maven/resources/pom-project.xml.template:
```diff
@@ -0,0 +1,35 @@
+    </properties>
+
+    <dependencyManagement>
+        <!--
+        You can centralize dependency information here.
+        See https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management
+        -->
```
I like the hints

mbien commented:
just as note: travis didn't trigger here, but there is no reason to panic (yet), it sometimes happens. It usually starts with the next update to the PR (e.g a rebase to delivery).

vieiro commented:
Hi,

I tried a 

```
$ git rebase delivery
Auto-fusionando java/libs.nbjavacapi/manifest.mf
CONFLICTO (contenido): Conflicto de fusión en java/libs.nbjavacapi/manifest.mf
error: no se pudo aplicar 8da225d7f7... Increment spec versions for Apache NetBeans 15 development.
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
No se pudo aplicar 8da225d7f7... Increment spec versions for Apache NetBeans 15 development.
```

But it seems there's a conflict in `java/libs.nbjavacapi/manifest.mf` (that is seems it's ready in master for NB15).

Wouldn't it be easier to cherry-pick this one?

On other news, I'm very consused with travis status and PHP editor failures! It this a known issue in Travis?

matthiasblaesing commented:
@vieiro yes a rebase would be very wrong in this case. A rebase would pull in all commits in your branch, that are not yet in the target branch and that is not what you want. You want to only pick your commit and thus yes a cherry-pick is right.

neilcsmith-net commented:
Sorry, rebase was a general reference to rebaseing the PR and base branch.

However, I generally do use the rebase command too eg. `git rebase HEAD~1 --onto delivery`

mbien commented:
> Wouldn't it be easier to cherry-pick this one?

yes ;) simply cherry pick it on top of latest delivery, then force push into your remote PR branch, then retarget this PR to delivery via the UI here (someone else can do this too).

> On other news, I'm very consused with travis status and PHP editor failures! It this a known issue in Travis?

I posted several mails regarding that to the dev list already, the job is simply too long (2.5h) which makes it automatically unreliable. It gonna be fine if restarted, but why waste CI time.

Regarding travis, yes travis has issues, it is what it is.

neilcsmith-net commented:
@vieiro thanks. Delivery is merged to master after every release candidate, so roughly weekly.

mbien commented:
i think there is an issue.
 - create a java maven application
 - create a "POM Project"

now i see the following:
```
tree
.
├── mavenproject1
│   ├── pom.xml
│   └── src
│       └── main
│           └── java
│               └── com
│                   └── mycompany
│                       └── mavenproject1
│                           └── Mavenproject1.java
└── pomproject1
    ├── pom_1.xml
    └── pom.xml
```
some paths must be wrong since the pom is copied into the wrong project and NB attaches a `_1`, the other pom is a default maven pom.

vieiro commented:
@mbien Verified this happens in Linux. Thanks! Will take a look at it in the EU evening...

sdedic reviewed (approved):
LGTM

vieiro commented:
This is somewhat difficult to debug (requires "Enabling J2SE" also).
It seems we're creating the pom.xml programmatically, this has the advantage of including a nested `<parent>` stuff in case we're nesting poms. An update is on the way (but till wake a day or so).

vieiro commented:
Hi all,

So I've removed the template: we do modify the pom.xml programmatically, i.e., a template is not good for us. We do this to add a `<parent>` stanza in Maven modules, for instance. A template for pom.xml won't fit us.

While at it I've also removed the "exec.mainClass" for non-jar packaging projects, that was being included in `<properties>` without need.

I'd appreciate a new review/testing.

mbien commented:
hi @vieiro, gave it a quick try and it worked.
```
tree pomproject1/
pomproject1/
└── pom.xml

0 directories, 1 file
```
one pom. The nice template with the hints isn't used but it fixes the bug - works for me :)

vieiro commented:
Hi @ebarboni , may we merge then?

mbien reviewed (approved):
code looks good as well. Tested other maven projects (e.g java and NB module) a bit just to make sure they still work and saw no issues.

Eric does the merges to delivery since he also has to sync things to other branches.

ebarboni commented:
merging this and baking RC6
